### PR TITLE
Replace node guard checks with logging in mempool peer messaging

### PR DESCRIPTION
**Pull request by mikedennis**
#635

### CODE REVIEW

**Comment by bokobza** [comment]
Looking at this.

File: Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs [hunk 1]
```diff
@@ -262,7 +262,13 @@ private Task ProcessMessageAsync(Node node, IncomingMessage message)
         /// <param name="message">The message payload.</param>
         private async Task SendMempoolPayloadAsync(Node node, MempoolPayload message)
         {
-            Guard.Assert(node == this.AttachedNode); // just in case
+            if (node != this.AttachedNode)
+            {
+                this.logger.LogError("Node Mismatch ({0}:'{1}', {2}:'{3}')", nameof(node), node?.RemoteSocketEndpoint, nameof(this.AttachedNode), this.AttachedNode?.RemoteSocketEndpoint);
```
Comment (bokobza) [inline Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: We should probably have a `Guard.NotNull(node, nameof(node));` in front of all these methods. We can then remove the conditonal access (`node?`).

**Comment by mikedennis** [comment]
Added the guards on node and downgraded the log from Error to Debug which is probably more appropriate.

File: Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs [hunk 2]
```diff
@@ -262,7 +262,14 @@ private Task ProcessMessageAsync(Node node, IncomingMessage message)
...
-            Guard.Assert(node == this.AttachedNode); // just in case
+            Guard.NotNull(node, nameof(node));
+            if (node != this.AttachedNode)
+            {
+                this.logger.LogDebug("Node Mismatch ({0}:'{1}', {2}:'{3}')", nameof(node), node.RemoteSocketEndpoint, nameof(this.AttachedNode), this.AttachedNode?.RemoteSocketEndpoint);
+                this.logger.LogTrace("(-)[NODE_MISMATCH]");
+                return;
+            }
+
             this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(node), node.RemoteSocketEndpoint, nameof(message), message.Command);
```
Comment (Aprogiena) [inline Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: the entry log should be preserved just after guard

then the if is OK, but should not be followed by entry log
it should be followed by log line such as
```
this.logger.LogDebug("Attached node '{0}' does not match the originating node '{1}'.", node.RemoteSocketEndpoint, this.AttachedNode?.RemoteSocketEndpoint);
```

same for the code below

**Comment by mikedennis** [comment]
Ok I think this should be good to go now.

**Pull request merged**